src/lib/common/string.cpp

```diff
@@ -791,6 +792,68 @@ bool str2double(const char* s, double* dP)
 
 
 
+/* ****************************************************************************
+*
+* decimalDigits
+*
+* This function counts the number of decimal digits of a given float, to a maximum of
+* PRECISION_DIGITS. The algorithm is inspired in http://stackoverflow.com/a/1083316/1485926
+* but with a "cutting condition" needed due to float representation may have an infinite
+* number of decimals, e.g. 3.14 could be internally coded as 3.1399999..
+*/
+unsigned int decimalDigits(double d)
+{
+  unsigned int digits = 0;
+
+  double intPart;
+  double decimalPart = fabs(modf(d, &intPart));
+
+  while (decimalPart > 0)
+  {
+    digits++;
+    decimalPart *= 10;
+    decimalPart = modf(decimalPart, &intPart);
+    if (fabs(1 - decimalPart ) < 0.01)
+    {
+      // In the case of 3.139999..., after step 2 modf() will produce 0.999..., so
```

Comment:
No opinion, not without empirical tests ....
floats is not an exact science ...

I just find it a little sad having to call two functions in the loop to get the number of decimals.
However, these two functions are probably implemented as macros anyways

NTC